hooks/tk-multi-publish2/basic/publish_session_geometry.py

```diff
@@ -0,0 +1,574 @@
+            "Publish file Template": {
+                "type": "template",
+                "default": None,
+                "description": "Template path for published work files. Should"
+                               "correspond to a template defined in "
+                               "templates.yml."
+            }
+        }
+
+    @property
+    def item_filters(self):
+        """
+        List of item types that this plugin is interested in.
+
+        Only items matching entries in this list will be presented to the
+        accept() method. Strings can contain glob patters such as *, for example
+        ["maya.*", "file.maya"]
+        """
+        return ["maya.session.geometry"]
+
+    def accept(self, settings, item):
+        """
+        Method called by the publisher to determine if an item is of any
+        interest to this plugin. Only items matching the filters defined via the
+        item_filters property will be presented to this method.
+
+        A publish task will be generated for each item accepted here. Returns a
+        dictionary with the following booleans:
+
+            - accepted: Indicates if the plugin is interested in this value at
+                all. Required.
+            - enabled: If True, the plugin will be enabled in the UI, otherwise
+                it will be disabled. Optional, True by default.
+            - visible: If True, the plugin will be visible in the UI, otherwise
+                it will be hidden. Optional, True by default.
+            - checked: If True, the plugin will be checked in the UI, otherwise
+                it will be unchecked. Optional, True by default.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+
+        :returns: dictionary with boolean keys accepted, required and enabled
+        """
+
+        accepted = True
+        publish_template = self._get_template("Publish file Template", settings)
+        if not publish_template:
+            accepted = False
+
+        return {
+            "accepted": accepted,
+            "checked": True
+        }
+
+    def validate(self, settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+
+        # check that the AbcExport command is available!
+        if not mel.eval("exists \"AbcExport\""):
+            return False
+        
+        # check that there is still geometry in the scene:
+        if not cmds.ls(geometry=True, noIntermediate=True):
```

Comment:
Nice. I think this should log an error as well so that the user can see why the validation failed. The message can be explicit so they know exactly how to correct it (by creating some geometry or turning off the plugin).